client/vrf/src/lib.rs

```diff
@@ -49,6 +50,10 @@ const LOG_TARGET: &str = "signing-vrf";
 
 /// Returns VRF pre-digest which includes an output signing the input info
 // TODO: get VrfInput via runtime API before calling this in client
+// TODO: get key via runtime API that gets it using the NimbusId that the client is using and the
+// KeysLookup in the runtime
+// TODO: where will this be called, in the client while authoring?
+// `nimbus_consensus::produce_candidate` seems like place where the PreDigest needs to be added
 pub fn vrf_predigest<Hash: AsRef<[u8]> + Clone>(
```

Comment:
This function forms the Vrf `PreDigest` which includes the `VrfOutput` and the `VrfProof` but not sure how to call it in `nimbus_consensus::produce_candidate` which seems to be the only place the inherent digests are added: https://github.com/PureStake/nimbus/blob/main/nimbus-consensus/src/lib.rs#L379

Maybe I can add this crate in moonbeam as a dependency to nimbus-consensus, but it feels wrong. Feels safer to move this crate to nimbus if that's the only place inherent digests are added.

Reply:
> From Elois:  "you need to create a DigestProvider trait somewhere in nimbus to allow nimbus users to inject their own digests. Then, in moonbeam we will implement this trait to inject our vrf digest"

So doing this now, seems like the right approach